test/evp_pkey_provided_test.c

```diff
@@ -550,6 +554,37 @@ static int test_fromdata_ec(void)
         || !TEST_int_eq(EVP_PKEY_size(pk), 2 + 35 * 2))
         goto err;
 
+    char pub[sizeof(ec_pub_keydata)];
+    char priv[sizeof(ec_priv_keydata)];
+    char curve_name[80];
+    OSSL_PARAM todata_params[4];
+    todata_params[0] =
+        OSSL_PARAM_construct_utf8_string(OSSL_PKEY_PARAM_EC_NAME, curve_name,
+                                         sizeof(curve_name));
+    todata_params[1] =
+        OSSL_PARAM_construct_octet_string(OSSL_PKEY_PARAM_PUB_KEY, pub,
+                                         sizeof(pub));
+    todata_params[2] =
+        OSSL_PARAM_construct_BN(OSSL_PKEY_PARAM_PRIV_KEY, (unsigned char *)priv,
+                                sizeof(priv));
+    todata_params[3] = OSSL_PARAM_construct_end();
+
+    if (!TEST_true(EVP_PKEY_todata(pk, todata_params)))
+        goto err;
+
+    if (!TEST_str_eq(curve_name, curve))
+        goto err;
+
+    /* The public key is returned is in COMPRESSED FORMAT */
+    if (!TEST_true(pub[0] == (POINT_CONVERSION_COMPRESSED + 1))
+        || !TEST_mem_eq(pub + 1, todata_params[1].data_size - 1,
+                        ec_pub_keydata + 1, compressed_sz - 1))
+        goto err;
+
+    /*
+     * TODO(3.0): How do we compare bignums here?
```

Comment:
Convert to BN and use the test macro?

Reply:
The data coming back is in reverse order - how does converting to BN help with this?

Reply:
`BN_native2bn` does the right thing.

Reply:
And create a BIGNUM with `BN_bin2bn` for `ec_priv_keydata`, 'cause I suppose that one is in BE order, right?